nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,536 @@
+
+          cp -f ${mainConfig} ${dataDir}/etc/sympa.conf
+          DBPASS="$(head -n1 ${cfg.database.passwordFile})"
+          if [ -n "$DBPASS" ]; then
+              sed -e "s,#dbpass#,$DBPASS,g" \
+                  -i ${dataDir}/etc/sympa.conf
+          else
+              sed -e "/db_passwd.*#dbpass#/d" \
+                  -i ${dataDir}/etc/sympa.conf
+          fi
+
+          ${concatStringsSep "\n" (flip mapAttrsToList cfg.domains (fqdn: domain:
+          ''
+            mkdir -p -m 750 ${dataDir}/etc/${fqdn}
+            cp -f ${robotConfig fqdn domain} ${dataDir}/etc/${fqdn}/robot.conf
+            mkdir -p -m 750 ${dataDir}/list_data/${fqdn}
+          ''))}
+
+          cp -f ${virtual} ${dataDir}/virtual.sympa
+          cp -f ${transport} ${dataDir}/transport.sympa
+          cp -f ${listAliases} ${dataDir}/etc/list_aliases.tt2
+
+          touch ${dataDir}/sympa_transport
+
+          ${optionalString (cfg.mta.type == "postfix") ''
+            ${pkgs.postfix}/bin/postmap hash:${dataDir}/virtual.sympa
+            ${pkgs.postfix}/bin/postmap hash:${dataDir}/transport.sympa
+          ''}
+          ${pkg}/bin/sympa_newaliases.pl
+
+
+          cp -af ${pkg}/static_content ${dataDir}/
+          # Yes, wwsympa needs write access to static_content..
+          chmod -R u+w,a+X ${dataDir}/static_content/
+          # Make static_content accessible to nginx
+          chmod o+X ${dataDir}
+
+          ${pkg}/bin/sympa.pl --health_check
+        '';
+      };
+      systemd.services.sympa-archive = {
+        description = "Sympa mailing list manager (archiving)";
+        bindsTo = [ "sympa.service" ];
+        restartTriggers = [ mainConfig ];
+        serviceConfig = sympaServiceConfig "archived";
+      };
+      systemd.services.sympa-bounce = {
+        description = "Sympa mailing list manager (bounce processing)";
+        bindsTo = [ "sympa.service" ];
+        restartTriggers = [ mainConfig ];
+        serviceConfig = sympaServiceConfig "bounced";
+      };
+      systemd.services.sympa-bulk = {
+        description = "Sympa mailing list manager (message distribution)";
+        bindsTo = [ "sympa.service" ];
+        restartTriggers = [ mainConfig ];
+        serviceConfig = sympaServiceConfig "bulk";
+      };
+      systemd.services.sympa-task = {
+        description = "Sympa mailing list manager (task management)";
+        bindsTo = [ "sympa.service" ];
+        restartTriggers = [ mainConfig ];
+        serviceConfig = sympaServiceConfig "task_manager";
+      };
+    }
+
+    (mkIf cfg.web.enable {
+      systemd.services.wwsympa = {
+        wantedBy = [ "multi-user.target" ];
+        after = [ "sympa.service" ];
+        restartTriggers = [ mainConfig ];
+        serviceConfig = {
+          Type = "forking";
+          Restart = "always";
+          ExecStart = ''${pkgs.spawn_fcgi}/bin/spawn-fcgi \
+            -u ${user} \
+            -g ${group} \
+            -U nginx \
```

Comment:
That's not a good idea when `cfg.web.server != "nginx"`

Reply:
Good catch, I wrapped the whole service in `if cfg.web.server == "nginx"` instead.